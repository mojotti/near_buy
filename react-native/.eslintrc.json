{
  "extends": "airbnb",
  "rules": {
    "func-names": ["error", "never"],
    "indent": ["error", 2],
    "arrow-parens": "off",
    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
    "react/jsx-indent": ["error", 2],
    "react/jsx-indent-props": ["error", 2],
    "import/no-named-as-default": 0,
    "import/prefer-default-export": 0,
    "no-undef": "off",
    "no-use-before-define": "off"
  },
  "env": {
    "jest": true
  },
  "globals": { "fetch": false }
}
