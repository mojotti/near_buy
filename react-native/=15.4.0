NearBuy@0.1.0 /Users/ristomatti/PycharmProjects/near_buy/react-native
└── UNMET PEER DEPENDENCY react@16.0.0

