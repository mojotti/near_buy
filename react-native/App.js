import React from 'react';
import { Provider } from 'react-redux';
import Application from './components/Application';
import store from './redux';

export default class App extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <Application />
      </Provider>
    );
  }
}
