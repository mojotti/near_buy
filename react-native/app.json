{
  "name": "NearBuy",
  "displayName": "NearBuy",
  "expo": {
    "sdkVersion": "25.0.0"
  }
}