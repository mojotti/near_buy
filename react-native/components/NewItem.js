'use strict';

import React, { Component } from 'react';
import store from '../redux';
import { Alert,
    Button,
    KeyboardAvoidingView,
    StyleSheet,
    TextInput,
    View } from 'react-native';
import { NavigationActions } from 'react-navigation';
import { localhost,
    widthWithThirtyPercentPadding} from "../src/static/constants";


export class NewItem extends Component {
    constructor(props) {
        super(props);
        this.state = {
            title: '',
            price: '',
            description: '',
        };
    }

    static navigationOptions = {
        title: 'Sell item',
    };

    getHeaders() {
        var headers = new Headers();
        headers.append('Content-Type', 'application/json');
        headers.append('Authorization', `Bearer ${store.getState().auth.token}`);
        return headers;
    }

    resetNavigationAndNavigateToRoute(targetRoute) {
        const resetAction = NavigationActions.reset({
            index: 0,
            actions: [
                NavigationActions.navigate({ routeName: targetRoute }),
            ],
        });
        this.props.navigation.dispatch(resetAction);
    }

    handleNewItemCreation () {
        const { title, price, description } = this.state;
        if (title == '' || price == '' || description == '') {
            Alert.alert('Invalid values', 'Enter at least price, title and description');
            return;
        }
        let url = `http://${localhost}:5000/api/v1.0/items`;
        fetch(url, {
            method: 'POST',
            body: JSON.stringify({
                'title': title,
                'price': price,
                'description': description
            }),
            headers: this.getHeaders()
        })
            .then((response) => response.json())
            .then((responseJson) => {
                console.log("pure " + responseJson.item.title);
                if (responseJson.item && responseJson.item.title === title) {
                    this.resetNavigationAndNavigateToRoute('Items');
                } else {
                    Alert.alert('Item creation failed', 'Something went wrong');
                }
            })
            .catch((error) => {
                console.error(error);
                return;
            });
    }

    onPriceChange(text) {
        let newText = '';
        let numbers = '0123456789';

        for (var i = 0; i < text.length; i++) {
            if ( numbers.indexOf(text[i]) > -1 ) {
                newText = newText + text[i];
            }
        }
        this.setState({price: newText})
    }

    render() {
        return (
            <View style={[styles.container]}>
                <TextInput
                    placeholder='Title'
                    autoCapitalize='sentences'
                    autoCorrect={false}
                    autoFocus={true}
                    maxLength = {20}
                    keyboardType='email-address'
                    value={this.state.title}
                    onChangeText={(text) => this.setState({ title: text })}
                    style={[styles.itemDetails]}
                />
                <TextInput
                    placeholder='Description'
                    autoCapitalize='sentences'
                    maxLength = {60}
                    autoCorrect={false}
                    autoFocus={false}
                    keyboardType='email-address'
                    value={this.state.description}
                    onChangeText={(text) => this.setState({ description: text })}
                    style={[styles.itemDetails]}
                />
                <KeyboardAvoidingView
                    behavior='padding'
                    keyboardVerticalOffset={164}
                >
                    <TextInput
                        placeholder='Price'
                        autoCorrect={false}
                        autoFocus={false}
                        keyboardType='numeric'
                        maxLength = {5}
                        value = {this.state.price}
                        onChangeText = {(text) => this.onPriceChange(text)}
                        style={[styles.itemDetails]}
                    />
                    <View style = {{paddingTop: 20}}></View>
                    <Button
                        onPress = {() => this.handleNewItemCreation()}
                        title = {'Submit item'}
                    />
                </KeyboardAvoidingView>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'space-around',
    },
    itemDetails: {
        width: widthWithThirtyPercentPadding,
        alignItems: 'center',
    }
});


export default NewItem;
