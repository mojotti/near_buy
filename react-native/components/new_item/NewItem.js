'use strict';

import React from 'react';
import store from '../../redux/index';
import {
  Alert,
  ScrollView,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import {
  alertInvalidValuesNewItem,
  localhost,
} from '../../src/static/constants';
import { styles } from '../../src/static/styles/NewItemStyles';
import { ItemDetails } from './ItemDetails';
import { ImageRow } from './ImageRow';

const UPPER_BUTTON_IDS = {
  leftButtonId: 0,
  rightButtonId: 1,
};

const LOWER_BUTTON_IDS = {
  leftButtonId: 2,
  rightButtonId: 3,
};

export class NewItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      price: '',
      description: '',
      mounted: false,
      images: [null, null, null, null],
    };

    this.handleTitleChange = this.handleTitleChange.bind(this);
    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
    this.handlePriceChange = this.handlePriceChange.bind(this);
    this.setNewImage = this.setNewImage.bind(this);
  }

  static navigationOptions = {
    title: 'Add item',
  };

  componentDidMount() {
    // work around for testing issue, should be solved when RN version is updated
    this.setState({ mounted: true });
  }

  getHeaders() {
    const headers = new Headers();
    headers.append('Content-Type', 'application/json');
    headers.append('Authorization', `Bearer ${store.getState().auth.token}`);
    return headers;
  }

  resetNavigationAndNavigateToRoute(targetRoute) {
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [NavigationActions.navigate({ routeName: targetRoute })],
    });
    this.props.navigation.dispatch(resetAction);
  }

  handleNewItemCreation() {
    const { title, price, description } = this.state;
    if (title === '' || price === '' || description === '') {
      Alert.alert(...alertInvalidValuesNewItem);
      return;
    }
    const url = `http://${localhost}:5000/api/v1.0/items`;
    fetch(url, {
      method: 'POST',
      body: JSON.stringify({
        title: title,
        price: price,
        description: description,
      }),
      headers: this.getHeaders(),
    })
      .then(response => response.json())
      .then(responseJson => {
        if (responseJson.item && responseJson.item.title === title) {
          this.resetNavigationAndNavigateToRoute('Items');
        } else {
          Alert.alert('Item creation failed', 'Something went wrong');
        }
      })
      .catch(error => {
        console.error(error);
      });
  }

  renderSubmitButton() {
    return (
      <TouchableHighlight
        id="SubmitButton"
        onPress={() => this.handleNewItemCreation()}
        style={styles.submitButton}
      >
        <Text style={styles.submitText}>{'Submit item'}</Text>
      </TouchableHighlight>
    );
  }

  setNewImage(image, id) {
    this.setState(prevState => {
      let images = prevState.images;
      images[id] = image;
      return images;
    });
    console.log('setting new image: ', image, id);
    console.log(this.state.images);
  }

  handleTitleChange(text) {
    this.setState({ title: text });
  }

  handleDescriptionChange(text) {
    this.setState({ description: text });
  }

  handlePriceChange(text) {
    this.setState({ price: text });
  }

  render() {
    const itemDetailsProps = {
      onTitleChange: this.handleTitleChange,
      onDescriptionChange: this.handleDescriptionChange,
      onPriceChange: this.handlePriceChange,
    };

    const firstImageRowProps = {
      onImageSelected: this.setNewImage,
      ...UPPER_BUTTON_IDS,
      images: this.state.images.slice(0, 2),
    };

    const secondImageRowProps = {
      onImageSelected: this.setNewImage,
      ...LOWER_BUTTON_IDS,
      images: this.state.images.slice(2),
    };

    return (
      <ScrollView>
        <View style={styles.container}>
          <Text style={styles.headerText}>Item details</Text>
          <ItemDetails {...itemDetailsProps} />
          <Text style={styles.headerText}>Add pictures</Text>
          <ImageRow {...firstImageRowProps} style={styles.imageRowMargin} />
          <ImageRow {...secondImageRowProps} style={styles.imageRowMargin} />
          {this.renderSubmitButton()}
        </View>
      </ScrollView>
    );
  }
}

export default NewItem;
