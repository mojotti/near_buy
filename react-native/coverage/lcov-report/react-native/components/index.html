<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for react-native/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> react-native/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.57% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>37/89</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.54% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>16/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.79% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>14/47</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.57% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>37/89</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Items.js"><a href="Items.js.html">Items.js</a></td>
	<td data-value="41.18" class="pic low"><div class="chart"><div class="cover-fill" style="width: 41%;"></div><div class="cover-empty" style="width:59%;"></div></div></td>
	<td data-value="41.18" class="pct low">41.18%</td>
	<td data-value="34" class="abs low">14/34</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="30" class="pct low">30%</td>
	<td data-value="20" class="abs low">6/20</td>
	<td data-value="41.18" class="pct low">41.18%</td>
	<td data-value="34" class="abs low">14/34</td>
	</tr>

<tr>
	<td class="file low" data-value="Login.js"><a href="Login.js.html">Login.js</a></td>
	<td data-value="41.82" class="pic low"><div class="chart"><div class="cover-fill" style="width: 41%;"></div><div class="cover-empty" style="width:59%;"></div></div></td>
	<td data-value="41.82" class="pct low">41.82%</td>
	<td data-value="55" class="abs low">23/55</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="20" class="abs medium">15/20</td>
	<td data-value="29.63" class="pct low">29.63%</td>
	<td data-value="27" class="abs low">8/27</td>
	<td data-value="41.82" class="pct low">41.82%</td>
	<td data-value="55" class="abs low">23/55</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Oct 18 2017 21:32:17 GMT+0300 (EEST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
