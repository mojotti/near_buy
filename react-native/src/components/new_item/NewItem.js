'use strict';

import React from 'react';
import store from '../../../redux/index';
import {
  Alert,
  ScrollView,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import {
  alertInvalidValuesNewItem,
  localhost,
} from '../../static/constants';
import { styles } from '../../static/styles/NewItemStyles';
import { ItemDetails } from './ItemDetails';
import { ImageRow } from './ImageRow';

const UPPER_BUTTON_IDS = {
  leftButtonId: 0,
  rightButtonId: 1,
};

const LOWER_BUTTON_IDS = {
  leftButtonId: 2,
  rightButtonId: 3,
};

export class NewItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      price: '',
      description: '',
      images: [null, null, null, null],
    };

    this.handleTitleChange = this.handleTitleChange.bind(this);
    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
    this.handlePriceChange = this.handlePriceChange.bind(this);
    this.setNewImage = this.setNewImage.bind(this);
  }

  static navigationOptions = {
    title: 'Add item',
  };

  getHeaders() {
    const headers = new Headers();
    headers.append('Content-Type', 'application/json');
    headers.append(
      'Authorization',
      `Bearer ${store.getState().authorizationReducer.token}`,
    );
    return headers;
  }

  resetNavigationAndNavigateToRoute(targetRoute) {
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [NavigationActions.navigate({ routeName: targetRoute })],
    });
    this.props.navigation.dispatch(resetAction);
  }

  handleNewItemCreation() {
    const { title, price, description } = this.state;
    if (title === '' || price === '' || description === '') {
      Alert.alert(...alertInvalidValuesNewItem);
      return;
    }
    const url = `http://${localhost}:5000/api/v1.0/items`;
    fetch(url, {
      method: 'POST',
      body: JSON.stringify({
        title: title,
        price: price,
        description: description,
      }),
      headers: this.getHeaders(),
    })
      .then(response => response.json())
      .then(responseJson => {
        this.handleResponse(responseJson);
      })
      .catch(error => {
        console.error(error);
      });
  }

  handleResponse(responseJson) {
    if (responseJson.item && responseJson.item.title === this.state.title) {
      this.resetNavigationAndNavigateToRoute('Items');
    } else {
      Alert.alert('Item creation failed', 'Something went wrong');
    }
  }

  renderSubmitButton() {
    return (
      <TouchableHighlight
        id="SubmitButton"
        onPress={() => this.handleNewItemCreation()}
        style={styles.submitButton}
      >
        <Text style={styles.submitText}>{'Submit item'}</Text>
      </TouchableHighlight>
    );
  }

  setNewImage(image, id) {
    this.setState(prevState => {
      let images = prevState.images;
      images[id] = image;
      return images;
    });
  }

  handleTitleChange(text) {
    this.setState({ title: text });
  }

  handleDescriptionChange(text) {
    this.setState({ description: text });
  }

  handlePriceChange(text) {
    this.setState({ price: text });
  }

  render() {
    const itemDetailsProps = {
      onTitleChange: this.handleTitleChange,
      onDescriptionChange: this.handleDescriptionChange,
      onPriceChange: this.handlePriceChange,
    };

    const firstImageRowProps = {
      onImageSelected: this.setNewImage,
      ...UPPER_BUTTON_IDS,
      images: this.state.images.slice(0, 2),
    };

    const secondImageRowProps = {
      onImageSelected: this.setNewImage,
      ...LOWER_BUTTON_IDS,
      images: this.state.images.slice(2),
    };

    return (
      <ScrollView>
        <View style={styles.container}>
          <Text style={styles.headerText}>Item details</Text>
          <ItemDetails {...itemDetailsProps} />
          <Text style={styles.headerText}>Add pictures</Text>
          <ImageRow {...firstImageRowProps} style={styles.imageRowMargin} />
          <ImageRow {...secondImageRowProps} style={styles.imageRowMargin} />
          {this.renderSubmitButton()}
        </View>
      </ScrollView>
    );
  }
}

export default NewItem;
