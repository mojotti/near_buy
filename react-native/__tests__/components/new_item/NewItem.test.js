import 'isomorphic-fetch'; // for headers, fetch, etc.;
import React from 'react';
import { Alert } from 'react-native';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import NewItem, { _NewItem } from '../../../src/components/new_item/NewItem';

jest.mock('react-navigation', () => {
  return {
    NavigationActions: {
      reset: jest.fn(() => {}),
      navigate: jest.fn(() => {}),
    },
  };
});

describe('<_NewItem />', () => {
  beforeEach(() => {
    Alert.alert = jest.fn();
  });

  test('renders correctly', () => {
    const newItem = renderer.create(<_NewItem />);

    expect(newItem.toJSON()).toMatchSnapshot();
  });

  test('calls handle new item when submit is pressed', () => {
    const handleNewItem = sinon.spy(
      _NewItem.prototype,
      'handleNewItemCreation',
    );

    const newItemComponent = shallow(<_NewItem />);

    newItemComponent
      .find('ScrollView')
      .find('TouchableHighlight')
      .simulate('Press');

    expect(handleNewItem.callCount).toEqual(1);
    handleNewItem.restore();
  });

  test('set new image changes state', () => {
    const newItemComponent = shallow(<_NewItem />);
    const newItemInstance = newItemComponent.instance();

    const image = { path: 'picture_of_me.png' };
    const index = 1;
    newItemInstance.setNewImage(image, index);

    expect(newItemComponent.state('images')[index]).toEqual(image);
  });

  test('sets state correctly', () => {
    const newItemComponent = shallow(<_NewItem />);
    const newItemInstance = newItemComponent.instance();

    const title = 'foo';
    const description = 'bar';
    const price = 666;

    newItemInstance.handleTitleChange(title);
    newItemInstance.handleDescriptionChange(description);
    newItemInstance.handlePriceChange(price);

    expect(newItemComponent.state('title')).toEqual(title);
    expect(newItemComponent.state('description')).toEqual(description);
    expect(newItemComponent.state('price')).toEqual(price);
  });

  test('gets correct header', () => {
    const props = {
      token: 'jippii',
    };
    const newItemComponent = shallow(<_NewItem {...props} />);
    const newItemInstance = newItemComponent.instance();

    const header = newItemInstance.getHeaders();
    expect(header).toMatchSnapshot();
  });

  test('handles new item creation', () => {
    fetch.mockResponseSuccess('foo');

    const newItemComponent = shallow(<_NewItem />);
    const newItemInstance = newItemComponent.instance();

    newItemComponent.setState({ title: 'title' });
    newItemComponent.setState({ description: 'description' });
    newItemComponent.setState({ price: 10 });

    newItemInstance.handleNewItemCreation();
  });

  test('handles new item creation response', () => {
    const navigateSpy = sinon.spy(
      _NewItem.prototype,
      'resetNavigationAndNavigateToRoute',
    );

    const navigation = { dispatch: jest.fn() };
    const newItemComponent = shallow(<_NewItem navigation={navigation} />);
    const newItemInstance = newItemComponent.instance();

    newItemComponent.setState({ title: 'title' });
    const response = {
      item: {
        title: 'title',
      },
    };
    newItemInstance.handleResponse(response);
    expect(navigateSpy.calledOnce).toBeTruthy();
  });

  test('handles new item creation response', () => {
    const newItemComponent = shallow(<_NewItem />);
    const newItemInstance = newItemComponent.instance();

    const response = {
      item: {
        title: 'title',
      },
    };

    newItemInstance.handleResponse(response);

    const expectedAlert = [['Item creation failed', 'Something went wrong']];

    expect(Alert.alert.mock.calls.length).toBe(1);
    expect(Alert.alert.mock.calls).toEqual(expectedAlert);
  });
});
