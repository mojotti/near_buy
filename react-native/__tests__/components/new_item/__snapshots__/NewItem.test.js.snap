// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<_NewItem /> images are added if they exist 1`] = `
Array [
  Object {
    "data": "12345",
    "filename": "image0.jpg",
    "name": "pictures[]",
    "type": "image/foo",
  },
  Object {
    "data": "12345",
    "filename": "image1.jpg",
    "name": "pictures[]",
    "type": "image/foo",
  },
]
`;

exports[`<_NewItem /> renders correctly 1`] = `
<RCTScrollView
  keyboardShouldPersistTaps="handled"
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "marginTop": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#848484",
            "fontFamily": null,
            "fontSize": 25,
            "marginBottom": 15,
            "marginLeft": 56.25,
            "marginTop": 20,
            "textAlign": "left",
          }
        }
      >
        Item details
      </Text>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 10,
              "elevation": 1,
              "height": 80.03999999999999,
              "marginBottom": 10,
              "width": 637.5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="sentences"
            autoCorrect={false}
            keyboardType="email-address"
            maxLength={40}
            onChangeText={[Function]}
            placeholder="Title"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "fontSize": 16,
                  "marginLeft": 15,
                  "marginRight": 15,
                  "padding": 5,
                  "width": 525,
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 10,
              "elevation": 1,
              "height": 266.8,
              "marginBottom": 10,
              "width": 637.5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="sentences"
            autoCorrect={false}
            keyboardType="email-address"
            maxLength={400}
            multiline={true}
            onChangeText={[Function]}
            placeholder="Description"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "fontSize": 16,
                  "marginLeft": 15,
                  "marginRight": 15,
                  "padding": 5,
                  "width": 525,
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 10,
              "elevation": 1,
              "height": 80.03999999999999,
              "marginBottom": 10,
              "width": 637.5,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCorrect={false}
            autoFocus={false}
            keyboardType="numeric"
            maxLength={5}
            onChangeText={[Function]}
            placeholder="Price"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "fontSize": 16,
                  "marginLeft": 15,
                  "marginRight": 15,
                  "padding": 5,
                  "width": 525,
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#848484",
            "fontFamily": null,
            "fontSize": 25,
            "marginBottom": 15,
            "marginLeft": 56.25,
            "marginTop": 20,
            "textAlign": "left",
          }
        }
      >
        Add pictures
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 5,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "padding": 8,
                },
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 5,
                },
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "elevation": 1,
                  "flex": 0.4,
                  "height": 262.5,
                  "justifyContent": "center",
                  "margin": 14,
                  "width": 262.5,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "gray",
                    "fontSize": 210.00000000000003,
                  },
                  Object {
                    "marginRight": 10,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 5,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "padding": 8,
                },
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 5,
                },
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "elevation": 1,
                  "flex": 0.4,
                  "height": 262.5,
                  "justifyContent": "center",
                  "margin": 14,
                  "width": 262.5,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "gray",
                    "fontSize": 210.00000000000003,
                  },
                  Object {
                    "marginRight": 10,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 5,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "padding": 8,
                },
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 5,
                },
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "elevation": 1,
                  "flex": 0.4,
                  "height": 262.5,
                  "justifyContent": "center",
                  "margin": 14,
                  "width": 262.5,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "gray",
                    "fontSize": 210.00000000000003,
                  },
                  Object {
                    "marginRight": 10,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 5,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "padding": 8,
                },
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 5,
                },
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 10,
                  "elevation": 1,
                  "flex": 0.4,
                  "height": 262.5,
                  "justifyContent": "center",
                  "margin": 14,
                  "width": 262.5,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "gray",
                    "fontSize": 210.00000000000003,
                  },
                  Object {
                    "marginRight": 10,
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4d4dff",
            "borderRadius": 30,
            "elevation": 1,
            "margin": 15,
            "marginTop": 15,
            "paddingBottom": 8,
            "paddingTop": 8,
            "width": 525,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": null,
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Submit item
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
